contracts/earn/strategies/Curve/StrategyCurveLP.sol

```diff
@@ -0,0 +1,241 @@
+import '@openzeppelin/contracts/utils/Pausable.sol';
+import '@uniswap/lib/contracts/libraries/TransferHelper.sol';
+
+import '../../../interfaces/TraderJoe/IUniswapRouter.sol';
+import '../../../interfaces/Curve/ICurveSwap.sol';
+import '../../../interfaces/Curve/IRewardsGauge.sol';
+import '../BaseStrategy.sol';
+
+contract StrategyCurveLP is BaseStrategy {
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    /* ----- State Variables ----- */
+
+    // Tokens used
+    address public crv;
+    address public native;
+    address public depositToken;
+
+    // Third party contracts
+    address public rewardsGauge;
+    address public pool;
+    uint256 public immutable poolSize;
+    uint256 public depositIndex;
+    bool public useUnderlying;
+
+    // Routes
+    address[] public crvToNativeRoute;
+    address[] public nativeToDepositRoute;
+
+    // if no CRV rewards yet, can enable later with custom router
+    bool public crvEnabled = true;
+    address public crvRouter;
+
+    address public uniRouter;
+
+    uint256 public lastHarvest;
+
+    /* ----- Events ----- */
+
+    event StratHarvest(address indexed harvester, uint256 wantHarvested, uint256 tvl);
+    event Deposit(uint256 tvl);
+    event Withdraw(uint256 tvl);
+
+    constructor(
+        address initVault,
+        address initAccessManager,
+        address initRewardsGauge,
+        address initPool,
+        uint256 initPoolSize,
+        uint256 initDepositIndex,
+        bool initUseUnderLying,
+        address[] memory initCrvToNativeRoute,
+        address[] memory initNativeToDepositRoute,
+        address initUniRouter
+    ) public BaseStrategy(initVault, initAccessManager) {
+        rewardsGauge = initRewardsGauge;
+        pool = initPool;
+        poolSize = initPoolSize;
+        depositIndex = initDepositIndex;
+        useUnderlying = initUseUnderLying;
+
+        crv = initCrvToNativeRoute[0];
+        native = initCrvToNativeRoute[initCrvToNativeRoute.length - 1];
+        crvToNativeRoute = initCrvToNativeRoute;
+        crvRouter = initUniRouter;
+
+        require(initNativeToDepositRoute[0] == native, 'StrategyCurveLP: initNativeToDepositRoute[0] != native');
+        depositToken = initNativeToDepositRoute[initNativeToDepositRoute.length - 1];
+        nativeToDepositRoute = initNativeToDepositRoute;
+        uniRouter = initUniRouter;
+
+        _giveAllowances();
+    }
+
+    /* ----- External Functions ----- */
+
+    function crvToNative() external view returns (address[] memory) {
+        return crvToNativeRoute;
+    }
+
+    function nativeToDeposit() external view returns (address[] memory) {
+        return nativeToDepositRoute;
+    }
+
+    /* ----- Public Functions ----- */
+
+    function harvest() public override whenNotPaused {
+        require(msg.sender == tx.origin || msg.sender == vault, 'StrategyCurveLP: !EOA or !Vault');
+
+        IRewardsGauge(rewardsGauge).claim_rewards(address(this));
```

Comment:
this line will claim all types of rewards ?   e.g. if the pool rewards wavax + crv + luna or wavax + crv + spell , it will claim all these rewards, right?